pydra/design/tests/test_python.py

```diff
@@ -0,0 +1,394 @@
+from operator import attrgetter
+import typing as ty
+from decimal import Decimal
+import attrs
+import pytest
+from pydra.engine.helpers import list_fields
+from pydra.engine.specs import PythonDef, PythonOutputs
+from pydra.design import python
+
+
+sort_key = attrgetter("name")
+
+
+def test_interface_wrap_function(tmp_path):
+    def func(a: int) -> float:
+        """Sample function with inputs and outputs"""
+        return a * 2
+
+    SampleDef = python.define(func)
+
+    assert issubclass(SampleDef, PythonDef)
+    inputs = sorted(list_fields(SampleDef), key=sort_key)
+    outputs = sorted(list_fields(SampleDef.Outputs), key=sort_key)
+    assert inputs == [
+        python.arg(name="a", type=int),
+        python.arg(name="function", type=ty.Callable, hash_eq=True, default=func),
```

Comment:
I know that probably it is not as common, but what if the argument is called "function"?

Reply:
It is a reserved name, it should raise an error when the task is defined

Reply:
Have implemented checks (and tests) for this